File: Team_work_Evan_Jairo/Typescript/Palindromo.ts
function shortestPalindrome(s: string): string {
    if (s === "") {
        return "";
    }

    const reverse = (str: string): string => str.split("").reverse().join("");

    if (reverse(s) === s) {
        return s;
    }

    const n = s.length;
    let aux2 = "";

    for (let i = 1; i < n; i++) {
        const aux = s.slice(n - i);
        aux2 = reverse(aux) + s;

        if (aux2 === reverse(aux2)) {
            break;
        }
    }

    return aux2;
}

File: Team_work_Evan_Jairo/Python/Palindromo.py
class Solution(object):
    def shortestPalindrome(self,s):
        if s == "":
            return ""

        if s == s[::-1]:
            return s
        
        n = len(s)
        for i in range(1,n):
            aux = s[n-i:n]
            aux2 = aux[::-1] + s

            if aux2 == aux2[::-1]:
                break

        return aux2
